java/util/HashMap/HashMap.hpp

```diff
@@ -24,504 +24,749 @@
+             * 						or null if none
+             */
+//            Value computeIfPresent (Key key,
+//                                  BiFunction<void (Key , Value, Value&)>
+//                                  mappingFunction) {
+//                Value oldValue = this->get(key);
+//                Value nullValue;
+//                Value newValue;
+//
+//                if (oldValue == nullValue) {
+//                    return nullValue;
+//                }
+//
+//                if (oldValue != nullValue) {
+//                    mappingFunction(key, oldValue, newValue);
+//
+//                    if (newValue != nullValue) {
+//                        this->replace(key, newValue);
+//
+//                        return newValue;
+//                    }
+//
+//                    if (newValue == nullValue) {
+//                        this->remove(key);
+//                        return nullValue;
+//                    }
+//                }
+//            }
+
+            /**
+             * Returns true if this map contains a mapping for the specified key.
+             *
+             * @param 	key 		The key whose presence in this map is to be tested
+             *
+             * @return 	true  		if this map contains a mapping for the specified key.
+             * 			false 		otherwise
+             */
+            boolean containsKey(const Key &key) const {
+                return this->original.find(key) != this->original.end();
+            }
+
+            /**
+             * Returns true if this map maps one or more keys to the specified value.
+             *
+             * @param 	value 		value whose presence in this map is to be tested
+             *
+             * @return 	true  		if this map maps one or more keys to the specified value
+             *  		false 		otherwise
+             */
+            boolean containsValue(const Value &value) const {
+                for (auto const &pair : this->original) {
+                    if (pair.second == value) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            /**
+             *  Set of hash map entries
+             *
+             * Returns a Set view of the mappings contained in this map.
+             * @return Set<Map.Entry<Key,Value>> - a set view of the mappings contained in this map
+             */
+            Set<class Map<Key, Value>::Entry> entrySet() {
+                Set<class Map<Key, Value>::Entry> entrySet;
+
+                for (auto const &pair : this->original) {
+                    Map<String, String>::Entry entry(pair.first, pair.second);
+                    entrySet.add(entry);
+                }
+
+                return entrySet;
+            }
+
+            /**
+             * Returns the value to which the specified key is mapped,
+             * or null if this map contains no mapping for the key

```

Comment:
default constructor or something, not null

Reply:
Thanks. Fixed.